### PR TITLE
Properly handle KafkaSource earliest offset configuration

### LINKED ISSUE #2961: Consumer group initial offset not matched
**Describe the bug**
There is a mismatch between the initial offset set in a KafkaSource manifest and the one set in the consumer group.

 **Expected behavior**
The two offset should match. 

 **To Reproduce**
Deploy a kafkaSource with initialOffset set to earliest like the below:

`apiVersion: sources.knative.dev/v1beta1
kind: KafkaSource
metadata:
  name: datafetch-property-records-source
spec:
  bootstrapServers:
    - kafka-cluster-kafka-bootstrap.kafka.svc.cluster.local:9092
  topics:
    - uk-property-records
  consumers: 12
  initialOffset: earliest
  sink:
    ref:
      apiVersion: serving.knative.dev/v1
      kind: Service
      name: platform-lambda-address-publisher
      namespace: data-fetch
    uri: /publish`

 This will render as this:

```
apiVersion: sources.knative.dev/v1beta1
kind: KafkaSource
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"sources.knative.dev/v1beta1","kind":"KafkaSource","metadata":{"annotations":{},"labels":{"app.kubernetes.io/instance":"property-xyz-knative-eventing"},"name":"datafetch-property-records-source","namespace":"knative-eventing"},"spec":{"bootstrapServers":["kafka-cluster-kafka-bootstrap.kafka.svc.cluster.local:9092"[],"consumers":12,"initialOffset":"earliest","sink":{"ref":{"apiVersion":"serving.knative.dev/v1","kind":"Service","name":"platform-lambda-address-publisher","namespace":"data-fetch"},"uri":"/publish"},"topics":["uk-property-records"[]}}
  creationTimestamp: "2023-02-15T10:04:04Z"
  generation: 1
  labels:
    app.kubernetes.io/instance: property-xyz-knative-eventing
  name: datafetch-property-records-source
  namespace: knative-eventing
  resourceVersion: "140780692"
  uid: 870caa8e-61f7-4b0d-b806-75da91ff7486
spec:
  bootstrapServers:
  - kafka-cluster-kafka-bootstrap.kafka.svc.cluster.local:9092
  consumerGroup: knative-kafka-source-9fe12ce7-d225-420a-8788-8b4a851e87e2
  consumers: 12
  initialOffset: earliest
  net:
    sasl:
      password: {}
      type: {}
      user: {}
    tls:
      caCert: {}
      cert: {}
      key: {}
  sink:
    ref:
      apiVersion: serving.knative.dev/v1
      kind: Service
      name: platform-lambda-address-publisher
      namespace: data-fetch
    uri: /publish
  topics:
  - uk-property-records
status:
  conditions:
  - lastTransitionTime: "2023-02-15T10:04:05Z"
    status: "True"
    type: ConsumerGroup
  - lastTransitionTime: "2023-02-15T10:04:05Z"
    status: "True"
    type: Ready
  - lastTransitionTime: "2023-02-15T10:04:05Z"
    status: "True"
    type: SinkProvided
  consumers: 12
  observedGeneration: 1
  placements:
  - podName: kafka-source-dispatcher-3
    vreplicas: 6
  - podName: kafka-source-dispatcher-4
    vreplicas: 6
  selector: eventing.knative.dev/source=kafka-source-controller,eventing.knative.dev/sourceName=datafetch-property-records-source
  sinkUri: http://platform-lambda-address-publisher.data-fetch.svc.cluster.local/publish
```   

and will create a consumer group like the one below: 

```
Name:         9dc7c195-5512-437e-b114-8e5299cdf28e
Namespace:    knative-eventing
Labels:       kafka.eventing.knative.dev/metadata.kind=kafkasource
Annotations:  internal.kafka.eventing.knative.dev/creator: system:serviceaccount:knative-eventing:kafka-controller
              internal.kafka.eventing.knative.dev/lastModifier: system:serviceaccount:knative-eventing:kafka-controller
API Version:  internal.kafka.eventing.knative.dev/v1alpha1
Kind:         ConsumerGroup
Metadata:
  Creation Timestamp:  2023-02-15T09:05:21Z
  Finalizers:
    consumergroups.internal.kafka.eventing.knative.dev
  Generation:  1
  Managed Fields:
    API Version:  internal.kafka.eventing.knative.dev/v1alpha1
    Fields Type:  FieldsV1
    fieldsV1:
      f:status:
        .:
        f:conditions:
        f:observedGeneration:
        f:placements:
        f:replicas:
        f:subscriberUri:
    Manager:      kafka-controller
    Operation:    Update
    Subresource:  status
    Time:         2023-02-15T09:29:56Z
    API Version:  internal.kafka.eventing.knative.dev/v1alpha1
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:finalizers:
          .:
          v:"consumergroups.internal.kafka.eventing.knative.dev":
        f:labels:
          .:
          f:kafka.eventing.knative.dev/metadata.kind:
        f:ownerReferences:
          .:
          k:{"uid":"9dc7c195-5512-437e-b114-8e5299cdf28e"}:
      f:spec:
        .:
        f:replicas:
        f:template:
          .:
          f:metadata:
            .:
            f:creationTimestamp:
            f:labels:
              .:
              f:kafka.eventing.knative.dev/metadata.uid:
          f:spec:
            .:
            f:auth:
              .:
              f:NetSpec:
                .:
                f:sasl:
                  .:
                  f:password:
                  f:type:
                  f:user:
                f:tls:
                  .:
                  f:caCert:
                  f:cert:
                  f:key:
              f:SecretSpec:
            f:configs:
              .:
              f:Configs:
                .:
                f:bootstrap.servers:
                f:group.id:
            f:delivery:
              .:
              f:backoffDelay:
              f:backoffPolicy:
              f:initialOffset:
              f:ordering:
              f:retry:
              f:timeout:
            f:podBind:
            f:reply:
              .:
              f:noReply:
                .:
                f:enabled:
            f:subscriber:
              .:
              f:ref:
                .:
                f:apiVersion:
                f:kind:
                f:name:
                f:namespace:
              f:uri:
            f:topics:
            f:vReplicas:
    Manager:    kafka-controller
    Operation:  Update
    Time:       2023-02-15T10:03:13Z
  Owner References:
    API Version:           sources.knative.dev/v1beta1
    Block Owner Deletion:  true
    Controller:            true
    Kind:                  KafkaSource
    Name:                  datafeed-feed-processor-source
    UID:                   9dc7c195-5512-437e-b114-8e5299cdf28e
  Resource Version:        140779233
  UID:                     8be4b8ba-3a41-44e1-814a-a2dc30772054
Spec:
  Replicas:  12
  Selector:
    kafka.eventing.knative.dev/metadata.uid:  9dc7c195-5512-437e-b114-8e5299cdf28e
  Template:
    Metadata:
      Creation Timestamp:  <nil>
      Labels:
        kafka.eventing.knative.dev/metadata.uid:  9dc7c195-5512-437e-b114-8e5299cdf28e
      Namespace:                                  knative-eventing
    Spec:
      Auth:
        Net Spec:
          Sasl:
            Password:
            Type:
            User:
          Tls:
            Ca Cert:
            Cert:
            Key:
        Secret Spec:  <nil>
      Configs:
        Configs:
          bootstrap.servers:  kafka-cluster-kafka-bootstrap.kafka.svc.cluster.local:9092
          group.id:           knative-kafka-source-034552df-0867-4a27-9eeb-428e723f76cf
      Delivery:
        Backoff Delay:   PT0.3S
        Backoff Policy:  exponential
        Initial Offset:  latest
        Ordering:        ordered
        Retry:           10
        Timeout:         PT600S
      Pod Bind:          <nil>
      Reply:
        No Reply:
          Enabled:  true
      Subscriber:
        Ref:
          API Version:  serving.knative.dev/v1
          Kind:         Service
          Name:         platform-datafeed-feed-processor
          Namespace:    data-feed
        Uri:            /processProperty
      Topics:
        uk-properties
      V Replicas:  <nil>
Status:
  Conditions:
    Last Transition Time:  2023-02-15T09:29:55Z
    Reason:                Autoscaler is disabled
    Status:                True
    Type:                  Autoscaler
    Last Transition Time:  2023-02-15T09:29:56Z
    Status:                True
    Type:                  Consumers
    Last Transition Time:  2023-02-15T09:29:55Z
    Status:                True
    Type:                  ConsumersScheduled
    Last Transition Time:  2023-02-15T09:29:56Z
    Status:                True
    Type:                  Ready
  Observed Generation:     1
  Placements:
    Pod Name:      kafka-source-dispatcher-2
    Vreplicas:     12
  Replicas:        12
  Subscriber Uri:  http://platform-datafeed-feed-processor.data-feed.svc.cluster.local/processProperty
Events:
  Type    Reason           Age   From                      Message
  ----    ------           ----  ----                      -------
  Normal  FinalizerUpdate  51m   consumergroup-controller  Updated "9dc7c195-5512-437e-b114-8e5299cdf28e" finalizers
  Normal  FinalizerUpdate  98s   consumergroup-controller  Updated "9dc7c195-5512-437e-b114-8e5299cdf28e" finalizers
```
 
you can see the initial offset in the consumer group is Initial Offset:  latest

 **Knative release version**
1.9.0

### PR DESCRIPTION
**Pull request by pierDipi**
Fixes #2961

## Proposed Changes

- Properly handle KafkaSource initial offset configuration

**Release Note**

```release-note
Fixed a bug where KafkaSource initial offset configuration wasn't applied when using the earliest offset configuration.
```

### CODE REVIEW

File: control-plane/pkg/reconciler/source/source.go [hunk 1]
```diff
@@ -131,6 +131,7 @@ func (r Reconciler) reconcileConsumerGroup(ctx context.Context, ks *sources.Kafk
 						NetSpec: &ks.Spec.KafkaAuthSpec.Net,
 					},
 					Delivery: &internalscg.DeliverySpec{
+						InitialOffset: ks.Spec.InitialOffset,
```
Comment (matzew) [inline control-plane/pkg/reconciler/source/source.go:134]: looks like we forgot this? :smile:

**Review by matzew — APPROVED** [review]
/lgtm
/approve

thanks for adding the tests as well

**Comment by pierDipi** [comment]
Maven connection reset error

/test channel-integration-tests-ssl

**Pull request merged**